### PR TITLE
Use pdebug queue for quartz allocation

### PR DESCRIPTION
**Pull request by bmhan12**
This PR:
- Targets pdebug queue for quartz. pbatch is rather popular.
- Fixes quartz not re-using `salloc` allocation
- `build_src.py` script now prints the name of the running node (hostname) for easier debugging in the future (Thanks @white238 for the suggestion!)

### CODE REVIEW

File: .gitlab/build_quartz.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ quartz_allocate:
   script:
     # Use when ellastic ci is on quartz or we go back to ruby
     #- salloc --reservation=ci --qos=ci_ruby -N 1 -c 36 -t 60 --no-shell --job-name=${PROJECT_ALLOC_NAME}
-    - salloc -N 1 -c 36 -t 60 --no-shell --job-name=${PROJECT_ALLOC_NAME}
+    - salloc -N 1 -c 36 -t 60 -p pdebug --no-shell --job-name=${PROJECT_ALLOC_NAME}
```
Comment (white238) [inline .gitlab/build_quartz.yml:31]: I looked at these jobs when I did the TOSS4 PR and I am not confident the `srun` 's are staying inside the outside `salloc`. It was allocating then sitting at the inner jobs waiting for more resources.

Comment (bmhan12) [reply .gitlab/build_quartz.yml:31]: > I looked at these jobs when I did the TOSS4 PR and I am not confident the `srun` 's are staying inside the outside `salloc`. It was allocating then sitting at the inner jobs waiting for more resources.

Think I saw the same thing.

`srun` failing to find existing allocation (no debug flag for `srun`):
https://lc.llnl.gov/gitlab/axom/axom/-/pipelines/290398

`srun` after adding debug flag:
https://lc.llnl.gov/gitlab/axom/axom/-/pipelines/290406

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ stages:
 .src_build_script:
   script:
     # Use pre-existing allocation if any
-    - JOBID=$(if [[ "$SYS_TYPE" == "toss_3_x86_64_ib" ]]; then squeue -h --name=${PROJECT_ALLOC_NAME} --format=%A; fi)
+    - export JOBID=$(if [[ "$SYS_TYPE" == "toss_4_x86_64_ib" ]]; then squeue -h --name=${PROJECT_ALLOC_NAME} --format=%A; fi)
```
Comment (bmhan12) [inline .gitlab-ci.yml:32]: This was the main change so `srun` would use the existing allocation.

**Review by agcapps — APPROVED** [review]
Thank you, @bmhan12

**Pull request merged**